src/engine.c

```diff
@@ -88,6 +93,21 @@ void draw_entity(Entity* entity, Shader uv_shader)
 // UV mapping, paralax rendering etc.
 void engine_draw_first_pass(Engine* engine, GameData* data)
 {
+	int i, x, y;
+
+	for (i = 0; i < data->room_count; i++)
+	{
+		Room* current_room = data->rooms[i];
+
+		for (y = 0; y < current_room->height; y++)
+			for(x = 0; x < current_room->width; x++)
+			{
+				Tile tile = current_room->tiles[y][x];
+				Rectangle source_rect = {tile.type * TILE_SIZE, 0, TILE_SIZE, TILE_SIZE};
+				DrawTextureRec(engine->texture_map, source_rect, (Vector2){x * TILE_SIZE + current_room->position.x, y * TILE_SIZE + current_room->position.y},WHITE);
```

Comment:
here you could move the vector2 our to reduce complexity

Reply:
OMG TRUER, THIS IS SOOOO TRUE